src/contracts/libraries/BytesArrayBitmaps.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity =0.8.12;
+
+/**
+ * @title Library for converting between an array of bytes and a bitmap.
+ * @author Layr Labs, Inc.
+ */
+library BytesArrayBitmaps {
+    /**
+     * @notice Converts an array of bytes into a bitmap.
+     * @param bytesArray The array of bytes to convert/compress into a bitmap.
+     * @return The resulting bitmap.
+     * @dev Each byte in the input is processed as indicating a single bit to flip in the bitmap
+     */
+    function bytesArrayToBitmap(bytes calldata bytesArray) internal pure returns (uint256) {
```

Comment:
Didn't understand this was what the non-ordered version would look like. Definitely cleaner, dump the ordered version :)

Reply:
wait until you see the Yul version